server/app/services/program/ProgramServiceImpl.java

```diff
@@ -690,6 +690,22 @@ public ProgramDefinition setBlockPredicate(
     return updateProgramDefinitionWithBlockDefinition(programDefinition, blockDefinition);
   }
 
+  @Override
+  @Transactional
+  public ProgramDefinition setBlockEligibilityDefinition(
+      long programId, long blockDefinitionId, @Nullable EligibilityDefinition eligibility)
```

Comment:
suggestion (blocking): can this take `Optional<EligibilityDefinition>` instead of being nullable?

Reply:
done